lru.go

```diff
@@ -28,6 +28,11 @@ func New[K comparable, V any](size int) (*Cache[K, V], error) {
 	return NewWithEvict[K, V](size, nil)
 }
 
+// NewSieve creates an LRU of the given size.
```

Comment:
should LRU be "SIEVE" here? Maybe we can add a reference too

Reply:
Sure. Will do!